lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -110,11 +110,41 @@ def add_customer_data(request, options, payment)
         }.compact
       end
 
-      def add_custom_fields_data(request, options)
-        if options[:device_fingerprint].present?
-          request[:DeviceFingerprint] = options[:device_fingerprint]
-          request[:ViaAgent] = false
+      def add_stored_credentials(request, options)
+        request[:PaymentModel] = add_payment_model(options)
+        if request[:PaymentModel].nil?
+          raise ArgumentError,
+            "stored credential options doesn't match with PaymentModels combination list, you can also pass in the options directly the PaymentModels value"
         end
+
+        if request[:PaymentModel].match?(/CIT-/)
+          request[:DeviceFingerprint] = options[:device_fingerprint] if options[:device_fingerprint]
+        end
+      end
+
+      def add_payment_model(options)
```

Comment:
💬  Comment:

Maybe the name should'nt be "add_payment_model" because you are not receiving the request body object to add more params.